File: src/main/java/com/zunza/buythedip/external/binance/dto/KlineRestApiResponse.java
package com.zunza.buythedip.external.binance.dto;

import java.math.BigDecimal;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.zunza.buythedip.external.binance.dto.deserializer.KlineDeserializer;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
@JsonDeserialize(using = KlineDeserializer.class)
public class KlineRestApiResponse {
	private Long openTime;
	private BigDecimal open;
	private BigDecimal high;
	private BigDecimal low;
	private BigDecimal close;
	private BigDecimal volume;
	private Long closeTime;
	private BigDecimal quoteAssetVolume;
	private Long numberOfTrades;
	private BigDecimal takerBuyBaseVolume;
	private BigDecimal takerBuyQuoteVolume;
	private BigDecimal ignore;
}

File: src/main/kotlin/com/zunza/buythedip/external/binance/dto/KlineRestApiResponse.kt
package com.zunza.buythedip.external.binance.dto

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.zunza.buythedip.external.binance.mapper.KlineDeserializer
import java.math.BigDecimal

@JsonDeserialize(using = KlineDeserializer::class)
data class KlineRestApiResponse(
    val openTime: Long,
    val open: BigDecimal,
    val high: BigDecimal,
    val low: BigDecimal,
    val close: BigDecimal,
    val volume: BigDecimal,
    val closeTime: Long,
    val quoteAssetVolume: BigDecimal,
    val numberOfTrades: Long,
    val takerBuyBaseVolume: BigDecimal,
    val takerBuyQuoteVolume: BigDecimal,
    val ignore: BigDecimal
)
